app/livechat/server/methods/returnAsInquiry.js

```diff
@@ -18,7 +19,9 @@ Meteor.methods({
 		if (!room.open) {
 			throw new Meteor.Error('room-closed', 'Room closed', { method: 'livechat:returnAsInquiry' });
 		}
+		const transferredBy = normalizeTransferredByData(Meteor.user() || {}, room);
```

Comment:
Since you already pass all the information you need through the `Livechat.returnRoomAsInquiry`, why not processing the `transferData` inside the `Livechat.returnRoomAsInquiry`? 
IMO it would reduce the complexity of the algorithm.